src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -888,6 +888,16 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
 
                 HttpResponseMessage responseMessage = WinHttpResponseParser.CreateResponseMessage(state, _doManualDecompressionCheck);
                 state.Tcs.TrySetResult(responseMessage);
+
+                if((responseMessage.StatusCode == HttpStatusCode.MultipleChoices ||
+                    responseMessage.StatusCode == HttpStatusCode.MovedPermanently ||
+                    responseMessage.StatusCode == HttpStatusCode.Redirect ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectMethod ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectKeepVerb) &&
```

Comment:
Should we consider status code 308: Permanent Redirect here?

Reply:
Maybe it would be less code/higher perf to just check for any 3xx code, i.e. do some simple math operation than listing all these out.

Reply:
I considered that but was concerned about `304 Not Modified` responses, which have a different behavior.